src/auth/executableresponse.ts

```diff
@@ -0,0 +1,166 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const SAML_SUBJECT_TOKEN_TYPE = 'urn:ietf:params:oauth:token-type:saml2';
+const OIDC_SUBJECT_TOKEN_TYPE1 = 'urn:ietf:params:oauth:token-type:id_token';
+const OIDC_SUBJECT_TOKEN_TYPE2 = 'urn:ietf:params:oauth:token-type:jwt';
+
+/**
+ * Interface defining JSON format of response from 3rd party executable run by
+ * pluggable auth client.
+ */
+export interface ExecutableResponseJson {
+  /**
+   * Version of the response, only version 1 currently supported.
+   * Always required.
+   */
```

Comment:
// The version of the JSON response. Only version 1 is currently supported.

Same below. I *think* it should be // and not /**

Reply:
is the // vs /** reasoning because the interface won't be exported externally or a different reason? I was a bit confused about what comment style to use in which contexts.

Reply:
I am not sure. I would consult the style guide

Reply:
From reading the docs, looks like the JSDoc style comments (/**) should be used for all top level exports from a module, so the interface + class here and the public properties. From that I should actually add comments for the properties in the ExecutableResponse class as well.